### PR TITLE
labels: Keep LabelArrays sorted

### PR DESCRIPTION
**Pull request by jrajahalme**
Create `labels.LabelArray` in sorted order which allows interacting with KV-store without worrying about label order. To be safe, all other uses of LabelArrays are also changed to sorted order, which affects e.g., rule labels. Arguably this is better than the somewhat arbitrary order they were in before.

This PR makes the follow-up work on `policy.SelectorCache` (#7539) easier. This also allows use of faster matching algorithms on LabelArrays in future.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7815)

### CODE REVIEW

**Comment by jrajahalme** [comment]
test-me-please

File: pkg/k8s/json_patch.go [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 package k8s
 
-const(
+const (
```
Comment (aanm) [inline pkg/k8s/json_patch.go]: There's a PR already for this change https://github.com/cilium/cilium/pull/7814

File: pkg/k8s/apis/cilium.io/utils/utils.go [hunk 2]
```diff
@@ -53,11 +53,12 @@ var (
 
 // GetPolicyLabels returns a LabelArray for the given namespace and name.
 func GetPolicyLabels(ns, name string, uid types.UID, derivedFrom string) labels.LabelArray {
```
Comment (aanm) [inline pkg/k8s/apis/cilium.io/utils/utils.go:55]: please add a unit test for this function to make sure the order forever is correct,

File: pkg/k8s/network_policy_test.go [hunk 3]
```diff
@@ -1640,10 +1640,10 @@ func (s *K8sSuite) TestGetPolicyLabelsv1(c *C) {
 		c.Assert(lbls, NotNil)
 		c.Assert(len(lbls), Equals, 4, Commentf(
 			"Incorrect number of labels: Expected Name, UID, Namespace and DerivedFrom labels."))
```
Comment (aanm) [inline pkg/k8s/network_policy_test.go]: You need to fix the comment

File: pkg/k8s/network_policy_test.go [hunk 4]
```diff
@@ -1640,10 +1640,10 @@ func (s *K8sSuite) TestGetPolicyLabelsv1(c *C) {
...
-		assertLabel(lbls[0], k8sConst.PolicyLabelName, tt.name)
-		assertLabel(lbls[1], k8sConst.PolicyLabelUID, tt.uuid)
+		assertLabel(lbls[0], k8sConst.PolicyLabelDerivedFrom, tt.derivedFrom)
```
Comment (aanm) [inline pkg/k8s/network_policy_test.go]: Do not use constants here, use their values instead, if we ever change the value of the constant we might change the order accidentally.

**Review by aanm — CHANGES_REQUESTED** [review]

File: pkg/k8s/json_patch.go (hunk 1 above)
Comment (jrajahalme) [reply pkg/k8s/json_patch.go]: dropping this once that merges

File: pkg/k8s/apis/cilium.io/utils/utils.go (hunk 2 above)
Comment (jrajahalme) [reply pkg/k8s/apis/cilium.io/utils/utils.go:55]: There are many already, adjusted them accordingly.

File: pkg/k8s/network_policy_test.go (hunk 4 above)
Comment (jrajahalme) [reply pkg/k8s/network_policy_test.go]: Changed.

File: pkg/k8s/network_policy_test.go (hunk 3 above)
Comment (jrajahalme) [reply pkg/k8s/network_policy_test.go]: Changed the order in the error message, too.

**Comment by jrajahalme** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

File: pkg/k8s/json_patch.go (hunk 1 above)
Comment (jrajahalme) [reply pkg/k8s/json_patch.go]: dropped this for the current version, as CI does not care about this :-)

**Review by aanm — APPROVED** [review]

**Pull request merged**